### ISSUE TITLE
Error handling for Tx/Rx syscalls is incorrect

### ISSUE THREAD

**Issue by jwbensley** [jwbensley/EtherateMT#44]
Due to being a massive dunce, the error handling for the various Tx and Rx syscalls being used is wrong....

This was highlighted in closed issue #19 

For example, `sendmsg()` below dies after one second, as soon a EtherateMT hits line rate `ENOBUFS` will be encountered, which is fine and there is no need to die/quit:

```
$ sudo taskset -c 3 ./build/etherate_mt -i enp1s0f0 -v -p2
Using inteface enp1s0f0 (6).
Verbose output enabled.
Setting interface promiscuous mode
Frame size set to 1514 bytes.
Using raw packet socket with sendmsg()/recvmsg().
Running in Tx mode.
Main thread pid is 11948.
Worker thread 11950 started
0.	Rx: 0.00 Gbps (0 fps) 0 Drops 0 Q-Freeze	Tx: 9.92 Gbps (818833 fps)	Err: 0
11950:Socket Tx error (105: No buffer space available)
Worker thread 11950 returned 1
Removing interface promiscuous mode
```

This applies to all syscall methods being used.

**Comment by jwbensley** [comment]
Fixed for packet send()/read() syscalls in commit: https://github.com/jwbensley/EtherateMT/commit/8da5a715c655aed50466f85cdb01b558ec63b6cc

**Comment by jwbensley** [comment]
Fixed for packet sendmsg()/recvmsg() syscalls in commit: https://github.com/jwbensley/EtherateMT/commit/8da5a715c655aed50466f85cdb01b558ec63b6cc

**Comment by jwbensley** [comment]
Fixed for TPACKET V2 send()/poll() syscalls in commits:
https://github.com/jwbensley/EtherateMT/commit/e0a0e3c72a5c7b57bf2660163927a9ecbe78df10
https://github.com/jwbensley/EtherateMT/commit/4ac3934ace558876dd65b78f26400b2824785417
https://github.com/jwbensley/EtherateMT/commit/55c9e442c0132daaadea7843904f4af6a78a3016

**Comment by jwbensley** [comment]
Fixed for `TPACKET_V3` `send()`/`poll()` syscalls in commit:
https://github.com/jwbensley/EtherateMT/commit/8ea0398897def904a8a6e88118144c95d4bd4f81

**Comment by jwbensley** [comment]
Code tidy in commit:
https://github.com/jwbensley/EtherateMT/commit/2ccaa4f3f7af614ffd907984700f92cbc411edb5

**Comment by jwbensley** [comment]
Fixed for `sendmmsg()`/`recvmmsg()` syscalls in commit:
https://github.com/jwbensley/EtherateMT/commit/285c78bf4d72ee7037b4ec3cc673c8e2fe555356

**Comment by jwbensley** [comment]
Now fixed with v1.0 release, merge #46
